service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ApplicantController.java

```diff
@@ -24,33 +31,70 @@
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        return applicantMigrationService.setMigratedValue(caseDetails);
+    }
 
     @SuppressWarnings("unchecked")
     @PostMapping("/mid-event")
     public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
         CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        ImmutableList.Builder<String> validationErrors = ImmutableList.builder();
 
-        Map<String, Object> applicantData = (Map<String, Object>) caseDetails.getData().get("applicant");
-        Applicant applicant = mapperService.mapObject(applicantData, Applicant.class);
-        List<String> validationErrors = new ArrayList<String>();
+        Map<String, Object> applicantData = (Map<String, Object>)
+            defaultIfNull(caseDetails.getData().get("applicant"), null);
+
+        if (caseDetails.getData().containsKey("applicants")) {
+            List<Map<String, Object>> migratedApplicantsObject =
+                (List<Map<String, Object>>) caseDetails.getData().get("applicants");
+
+            migratedApplicantsObject.stream()
+                .map(applicant ->
+                    mapperService.mapObject((Map<String, Object>) applicant.get("value"), Applicant.class))
+                .map(Applicant::getParty)
+                .map(ApplicantParty::getPbaNumber)
+                .filter(Objects::nonNull)
+                .forEach(pbaNumber -> {
+                    String newPbaNumberData = PBANumberHelper.updatePBANumber(pbaNumber);
+                    validationErrors.addAll(validatePBANumber(newPbaNumberData));
+                });
+
+            caseDetails.getData().put("applicants", migratedApplicantsObject);
 
-        if (applicant.getPbaNumber() == null || applicant.getPbaNumber().isBlank()) {
-            return AboutToStartOrSubmitCallbackResponse.builder()
-                .data(caseDetails.getData())
-                .errors(validationErrors)
-                .build();
         } else {
+            OldApplicant applicant = mapperService.mapObject(applicantData, OldApplicant.class);
+
+            if (applicant.getPbaNumber() == null || applicant.getPbaNumber().isBlank()) {
+                return AboutToStartOrSubmitCallbackResponse.builder()
+                    .data(caseDetails.getData())
+                    .errors(validationErrors.build())
+                    .build();
+            }
+
             String newPbaNumberData = PBANumberHelper.updatePBANumber(applicant.getPbaNumber());
-            validationErrors = PBANumberHelper.validatePBANumber(newPbaNumberData);
-            if (validationErrors.isEmpty()) {
+            validationErrors.addAll(validatePBANumber(newPbaNumberData));
+
+            if (validationErrors.build().isEmpty()) {
                 applicantData.put("pbaNumber", newPbaNumberData);
```

Comment:
That logic is missing when `applicants` (new structure) exists which means no value correction will be there anymore.

Regression!

Reply:
@swalker125 what does this mean?

Reply:
its on line 66-68 no?

Reply:
yep logic checks between new and old using an if and else block

Reply:
is there tests to highlight this?

Reply:
yes in the pbanumberhelper tests and i test in mideventcontroller also